server.go

```diff
@@ -657,15 +657,42 @@ func wireToAddrmgrNetAddresses(netAddr []*wire.NetAddress) []*addrmgr.NetAddress
 	return addrs
 }
 
+// wireToAddrmgrNetAddressesV2 converts a collection of version 2 wire network
+// addresses to a collection of address manager network addresses.  If any
+// addresses are not able to be converted, an error is returned.
+func wireToAddrmgrNetAddressesV2(netAddr []*wire.NetAddressV2) ([]*addrmgr.NetAddress, error) {
+	addrs := make([]*addrmgr.NetAddress, len(netAddr))
+	for i, wireAddr := range netAddr {
+		addrmgrAddrType := addrmgr.NetAddressType(wireAddr.Type)
+		addr, err := addrmgr.NewNetAddressByType(addrmgrAddrType, wireAddr.IP,
+			wireAddr.Port, wireAddr.Timestamp, wireAddr.Services)
+		if err != nil {
+			return nil, err
+		}
+		addrs[i] = addr
+	}
+	return addrs, nil
+}
+
 // addrmgrToWireNetAddress converts an address manager net address to a wire net
 // address.
 func addrmgrToWireNetAddress(netAddr *addrmgr.NetAddress) *wire.NetAddress {
 	return wire.NewNetAddressTimestamp(netAddr.Timestamp, netAddr.Services,
 		netAddr.IP, netAddr.Port)
 }
 
+// addrmgrToWireNetAddressV2 converts an address manager net address to a wire net
+// address.
+func addrmgrToWireNetAddressV2(netAddr *addrmgr.NetAddress) *wire.NetAddressV2 {
+	wireNetAddrType := wire.NetAddressType(netAddr.Type)
```

Comment:
There should probably be a conversion func instead of a direct cast so that changing their respective absolute values doesn't break this.